tools/pubsys/src/repo.rs

```diff
@@ -425,17 +425,33 @@ fn get_signing_key_source(signing_key_config: &SigningKeyConfig) -> Result<Box<d
     }
 }
 
-/// Helper function that generations KMSClient with region (or None) given config containing available keys
-fn get_client(config: &KMSKeyConfig, key_id: &String) -> Result<Option<KmsClient>> {
-    if let Some(region) = config.available_keys.get(key_id) {
-        Ok(Some(KmsClient::new(
-            Region::from_str(region).context(error::ParseRegionSnafu { what: region })?,
-        )))
+/// Helper function that generates a KmsClient or None given config containing available keys
+fn get_client(kmskey_config: &KMSKeyConfig, key_id: &str) -> Result<Option<KmsClient>> {
+    if let Some(region) = kmskey_config.available_keys.get(key_id) {
+        // We are cloning this so that we can send it across a thread boundary
+        let region = region.to_owned();
+        // We need to spin up a new thread to deal with the async nature of the
+        // AWS SDK Rust
+        thread::spawn(move || {
+            let runtime = tokio::runtime::Runtime::new().context(error::RuntimeCreationSnafu)?;
+            Ok(Some(runtime.block_on(async_get_client(&region))))
+        })
+        .join()
+        .map_err(|_| error::Error::ThreadJoin {})?
```

Comment:
I'd prefer to handle this the same way that `main.rs` handles conditionally creating a Tokio runtime around certain subcommands.